compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -128,19 +201,30 @@ MaterializeEncodingInfo getEncodingInfoForMatmul(EncodingAttr encoding,
                                                  TileMxNxK tileMxNxK) {
   EncodingUser user = encoding.getUser().getValue();
   EncodingRole role = encoding.getRole().getValue();
+  auto perm = getPermutationToCanonicalMatmulShape(encoding).value();
   bool isVector = isVectorEncoding(rank, user);
   bool isVecmatVector = (isVector && (isVecmatEncoding(encoding) ||
                                       isBatchVecmatEncoding(encoding)));
   bool isMatvecVector = (isVector && (isMatvecEncoding(encoding) ||
                                       isBatchMatvecEncoding(encoding)));
   // Start dim of the MxK (LHS), KxN (RHS), or MxN (RESULT) 2D matrix.
   int64_t matmulDimBase = isBatchMatmulEncodingUser(user) ? 1 : 0;
-
   MaterializeEncodingInfo encodingInfo;
-  if (isVector) {
-    encodingInfo.innerDimsPos = {matmulDimBase};
-  } else {
-    encodingInfo.innerDimsPos = {matmulDimBase, matmulDimBase + 1};
+  encodingInfo.innerDimsPos = {perm[matmulDimBase]};
+  if (!isVector) {
+    encodingInfo.innerDimsPos.push_back(perm[matmulDimBase + 1]);
+    encodingInfo.outerDimsPerm =
+        llvm::to_vector(llvm::seq<int64_t>(0, matmulDimBase));
+    encodingInfo.outerDimsPerm.push_back(matmulDimBase);
+    encodingInfo.outerDimsPerm.push_back(matmulDimBase + 1);
+    encodingInfo.outerDimsPerm =
+        applyPermutation(perm, encodingInfo.outerDimsPerm);
+  } else if (matmulDimBase > 0) {
+    encodingInfo.outerDimsPerm =
+        llvm::to_vector(llvm::seq<int64_t>(0, matmulDimBase));
+    encodingInfo.outerDimsPerm.push_back(matmulDimBase);
+    encodingInfo.outerDimsPerm =
+        applyPermutation(perm, encodingInfo.outerDimsPerm);
   }
```

Comment:
shouldn't we set outerDimsPerm based on indexing_maps? Why only these two cases are considered? What happens to `else`? If we only want to support cases less than or equal to 4D, can you add a comment?